src/main/java/de/hhu/stups/plues/ui/controller/Reports.java

```diff
@@ -185,67 +270,242 @@ private void displayModuleAbstractUnitUnitSemesterConflicts(final Store store,
+
+    return mandatoryModules;
   }
 
-  private void displayImpossibleCourses(final Store store, final ReportData reportData) {
-    impossibleCourses.setData(reportData.getImpossibleCourses()
-          .stream().map(store::getCourseByKey).collect(Collectors.toList()),
+  private List<List<Course>> displayImpossibleCourses(final Store store,
+                                                                 final ReportData reportData) {
+    List<Course> impossibleCourses = reportData.getImpossibleCourses()
+        .stream().map(store::getCourseByKey).collect(Collectors.toList());
+    List<Course> impossibleCoursesBecauseOfImpossibleModules =
         reportData.getImpossibleCoursesBecauseofImpossibleModules()
-          .stream().map(store::getCourseByKey).collect(Collectors.toList()),
+          .stream().map(store::getCourseByKey).collect(Collectors.toList());
+    List<Course> impossibleCoursesBecauseOfImpossibleModuleCombinations =
         reportData.getImpossibleCoursesBecauseOfImpossibleModuleCombinations()
-          .stream().map(store::getCourseByKey).collect(Collectors.toList()));
+          .stream().map(store::getCourseByKey).collect(Collectors.toList());
+    this.impossibleCourses.setData(impossibleCourses,
+        impossibleCoursesBecauseOfImpossibleModules,
+        impossibleCoursesBecauseOfImpossibleModuleCombinations);
+
+    return Arrays.asList(impossibleCourses, impossibleCoursesBecauseOfImpossibleModules,
+        impossibleCoursesBecauseOfImpossibleModuleCombinations);
   }
 
-  private void displayImpossibleModules(final Store store, final ReportData reportData) {
-    impossibleModules.setData(reportData.getIncompleteModules()
-        .stream().map(store::getModuleById).collect(Collectors.toList()),
+  private List<List<Module>> displayImpossibleModules(final Store store,
+                                                                 final ReportData reportData) {
+    List<Module> incompleteModules = reportData.getIncompleteModules()
+        .stream().map(store::getModuleById).collect(Collectors.toList());
+    List<Module> impossibleModulesBecauseOfMissingElectiveAbstractUnits =
         reportData.getImpossibleModulesBecauseOfMissingElectiveAbstractUnits()
-        .stream().map(store::getModuleById).collect(Collectors.toList()));
+        .stream().map(store::getModuleById).collect(Collectors.toList());
+    impossibleModules.setData(incompleteModules,
+        impossibleModulesBecauseOfMissingElectiveAbstractUnits);
+
+    return Arrays.asList(incompleteModules,
+        impossibleModulesBecauseOfMissingElectiveAbstractUnits);
+  }
+
+  private List<Unit> displayUnitsWithoutAbstractUnits(final Store store) {
+    List<Unit> unitsWithoutAbstractUnits = store.getUnits().stream()
+        .filter(unit -> unit.getAbstractUnits().size() == 0).collect(Collectors.toList());
+    this.unitsWithoutAbstractUnits.setData(unitsWithoutAbstractUnits);
+
+    return unitsWithoutAbstractUnits;
```

Comment:
split into a method that computes the data (e.g. in RealReportData) and one step that displays it.